### PR TITLE
[FIX] account: remove no longer applicable tax lines

### PR DESCRIPTION
**Pull request by nicocrm**
Behavior prior to this commit:

- if the tax eligibility is changed between the time the invoice created and the time it is posted, the invoice ends up with 2 tax lines: one for the "Tax Received" account and one for the "Cash Basis Transition" account
- there was already code intended to remove such lines, at https://github.com/odoo/odoo/blob/36b7fbe75dc45cc98a698c4af610ee4a5114919e/addons/account/models/account_move.py#L642, but it was not hit when the `recompute_tax_base_amount` parameter was true (something that was added in 226a1155124e890d00e33968feb43b9fd6b99fa5)

Behavior after this commit:
- only the tax line that is still applicable is kept on the invoice

opw-2371745

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by ngaller** [comment]
@fw-bot ignore 

This works OK in 14.0

**Comment by ngaller** [comment]
@nim-odoo ready for review

**Review by nim-odoo — COMMENTED** [review]
Changing a tax while it is already in use is rarely a good idea (even in a draft invoice). The fix looks fine, but could you please check if the use case reported in 226a1155124e890d00e33968feb43b9fd6b99fa5 is still working correctly? If it is not too much hassle, you could also add a test for this one.

I'd forward-port the test up to master, though.

**Comment by ngaller** [comment]
@fw-bot up to master

We can still port the tests

**Comment by ngaller** [comment]
@nim-odoo I added the test, in so doing I realized the line from 226a1155124e890d00e33968feb43b9fd6b99fa5 may actually have become superfluous... because the next `else` branch still sets the tax base amount (so my test still passes, even with the line commented out).  In effect this behaves as though `recompute_tax_base_amount` is always true, **except** when `recompute_tax_base_amount` we do not update the other  3 fields: `amount_currency`, `debit`, and `credit`. 

Nevertheless, commenting out the `recompute_tax_base_amount` branch makes another test fail.  So it seems more cautious to leave it untouched!

**Review by nim-odoo — APPROVED** [review]
robodoo delegate+

(note: I would wait until tomorrow morning to merge this)

**Comment by ngaller** [comment]
@robodoo r+

**Pull request closed without merging**
